pkg/search/postgres/common.go

```diff
@@ -897,6 +897,66 @@ func RunCursorQueryForSchema(ctx context.Context, schema *walker.Schema, q *v1.Q
 	}, closer, nil
 }
 
+// RunCursorQueryForSchemaType creates a cursor against the database
+func RunCursorQueryForSchemaType[T any, PT interface {
+	Unmarshal([]byte) error
+	*T
+}](ctx context.Context, schema *walker.Schema, q *v1.Query, db *pgxpool.Pool) (fetcher func(n int) ([]*T, error), closer func(), err error) {
+	query, err := standardizeQueryAndPopulatePath(q, schema, GET)
```

Comment:
There's a lot of code duplication here. Can you at least factor out the cursor creation to a common function?

Reply:
Extracted common code